src/vs/workbench/api/browser/mainThreadFileSystem.ts

```diff
@@ -158,12 +161,14 @@ export class MainThreadFileSystem implements MainThreadFileSystemShape {
 		return this._fileService.activateProvider(scheme);
 	}
 
-	$watch(session: number, resource: UriComponents, opts: IWatchOptions): void {
+	$watch(extension: IExtensionDescription, session: number, resource: UriComponents, opts: IWatchOptions): void {
```

Comment:
Be care with the description object - it can be huge because it includes all of package.json. It's better to send the id or name only

Reply:
@jrieken good point, I fixed that in https://github.com/microsoft/vscode/pull/139881/commits/506f19d56e3d6ccd4b23e6bb05b641b0ecfddae4

Btw I see more locations where we seem to send over the entire descriptor:

![image](https://user-images.githubusercontent.com/900690/148506636-9b1a4702-d175-4e2f-b4d3-bb49b302de72.png)

Reply:
https://github.com/microsoft/vscode/commit/913952ac2ea3c313a49e2951375f4c9967c79a5c takes care of the progress one

Reply:
d528769c4d55b3b7c271e3a75c942c77794f7522 for the messages